modules/tiles/src/tileset/helpers/memory-utils.ts

```diff
@@ -0,0 +1,22 @@
+import {GLTF} from '@loaders.gl/gltf';
+
+/**
+ * Calculate the GPU memory used by a GLTF tile, for both buffer and texture memory
+ * @param gltf - the gltf content of a GLTF tile
+ * @returns - total memory usage in bytes
+ */
+export function getGltfMemoryUsage(gltf: GLTF) {
+  let {images, bufferViews} = gltf;
+  images = images || [];
+  bufferViews = bufferViews || [];
+  const imageBufferViews = images.map((i) => i.bufferView);
```

Comment:
That's why I went with 4 bytes per pixel rather than reading the buffer size. As the images I'm working with are jpeg, the buffer size is too small. You're right I didn't consider mipmaps, do you think it is worth adding 33% on top to account for this?

Reply:
Yes that works for JPEG case. However if the tiles are using basis compressed textures (which they should) the GPU/CPU size is 1-to-1  so we would massively overcount texture memory use (probably with a factor of 4 or more). (though note that for compressed textures all mipmap levels are provided rather than autogenerated an must be summed up)

I have a vague recollection that mipmap memory is laid out for access efficiency and not for maximum compactness and that a factor of 1.5 should be used for at least some GPUs. But I guess that is splitting hairs, you may want to define a constant, choose a value and add a comment along these lines.